// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: order/v1/order.proto

package orderv1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on GetMerchantOrdersReq with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetMerchantOrdersReq) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetMerchantOrdersReq with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetMerchantOrdersReqMultiError, or nil if none found.
func (m *GetMerchantOrdersReq) ValidateAll() error {
	return m.validate(true)
}

func (m *GetMerchantOrdersReq) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for MerchantId

	// no validation rules for Page

	// no validation rules for PageSize

	if len(errors) > 0 {
		return GetMerchantOrdersReqMultiError(errors)
	}

	return nil
}

// GetMerchantOrdersReqMultiError is an error wrapping multiple validation
// errors returned by GetMerchantOrdersReq.ValidateAll() if the designated
// constraints aren't met.
type GetMerchantOrdersReqMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetMerchantOrdersReqMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetMerchantOrdersReqMultiError) AllErrors() []error { return m }

// GetMerchantOrdersReqValidationError is the validation error returned by
// GetMerchantOrdersReq.Validate if the designated constraints aren't met.
type GetMerchantOrdersReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetMerchantOrdersReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetMerchantOrdersReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetMerchantOrdersReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetMerchantOrdersReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetMerchantOrdersReqValidationError) ErrorName() string {
	return "GetMerchantOrdersReqValidationError"
}

// Error satisfies the builtin error interface
func (e GetMerchantOrdersReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetMerchantOrdersReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetMerchantOrdersReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetMerchantOrdersReqValidationError{}
